src/base.ts

```diff
@@ -3,7 +3,7 @@ import { MetadataFromDocument, MetricMetadata, NewRelicSink, Recorder, Sink, Std
 import { Parser } from '@asyncapi/parser';
 import { Specification } from 'models/SpecificationFile';
 import { join, resolve } from 'path';
-import { existsSync } from 'fs-extra';
+import { existsSync, statSync } from 'fs-extra';
```

Comment:
Why not using the asynchronous versions of these functions? 🤔 You can easily make `setSource` an async function and then it would be easy to call `await stat()` instead.

Reply:
Working fine with: `
const stats = await fPromises.stat(specFilePath);
`
Don't know if there is any better alternative to make `setSource` an async function.

Reply:
@peter-rr I don't think there is any. we either have to use promises and `await` it. or use Sync functions. which I guess the first option is a lot better.

Reply:
LGTM 👍